images/linux/scripts/installers/java-tools.sh

```diff
@@ -69,20 +69,18 @@ ln -s /usr/share/apache-maven-3.8.1/bin/mvn /usr/bin/mvn
 
 # Install Gradle
 # This script downloads the latest HTML list of releases at https://gradle.org/releases/.
-# Then, it extracts the top-most release download URL, relying on the top-most URL being for the latest release.
-# The release download URL looks like this: https://services.gradle.org/distributions/gradle-5.2.1-bin.zip
-# The release version is extracted from the download URL (i.e. 5.2.1).
+# Then, it extracts the latest release version and the release url for this version.
 # After all of this, the release is downloaded, extracted, a symlink is created that points to it, and GRADLE_HOME is set.
 wget -qO gradleReleases.html https://gradle.org/releases/
-gradleUrl=$(grep -m 1 -o "https:\/\/services.gradle.org\/distributions\/gradle-.*-bin\.zip" gradleReleases.html | head -1)
-gradleVersion=$(echo $gradleUrl | sed -nre 's/^[^0-9]*(([0-9]+\.)*[0-9]+).*/\1/p')
+gradleLatestVersion=$(grep -oP "gradle-\K.*(?=-bin\.zip)" gradleReleases.html | sort -V | tail -n1)
```

Comment:
I think it will be easier to parse : https://github.com/gradle/gradle/releases

Reply:
I decided to parse the release page since versions on GitHub appear earlier. Right now:
![image](https://user-images.githubusercontent.com/48208649/118274430-ccab3600-b4cd-11eb-953b-00fec0a04642.png)
but the releases only have 7.0.1
![image](https://user-images.githubusercontent.com/48208649/118274415-c5842800-b4cd-11eb-8e7f-5680e04d9140.png)

Reply:
Link works - https://services.gradle.org/distributions/gradle-7.0.2-bin.zip

Reply:
Good choice - https://services.gradle.org/versions/current

Reply:
wow, nice find! I wonder what was there when version 6.9 was released after 7.0. Did that page stay on 7 or switch to 6.9? That's one of the reasons why I'd like to retrieve all the versions from the releases page and sort them to get the latest major release

Reply:
I think it's better to use this json https://services.gradle.org/versions/all and take the latest version manually